### PR TITLE
Fix crash when FCD frame has bad length but CRC is OK.

### PR DESCRIPTION
**Pull request by dragos-oancea**

### CODE REVIEW

**Comment by coppice-git** [comment]
That fix is silently ignoring the reception of a good (i.e. the CRC is OK) frame with bad contents, which should not happen. A better fix is something like:

    if (len < 4)
    {
        span_log(&s->logging, SPAN_LOG_FLOW, "Bad length for FCD frame - %d\n", len);
        /* This frame didn't get corrupted in transit, because its CRC is OK. It was sent bad
           and there is little possibility that causing a retransmission will help. It is best
           to just give up. */
        t30_set_status(s, T30_ERR_TX_ECMPHD);
        terminate_call(s);
        return;
    }
    /*endif*/

at the start of process_rx_fcd(), to record that the erroneous from was received, and handle it similarly to other bad frames

**Comment by dragos-oancea** [comment]
the fix is for: https://github.com/freeswitch/spandsp/issues/38 . it's more like a "hotfix". we will try your approach.

File: src/t30.c [hunk 1]
```diff
@@ -3582,7 +3593,7 @@ static void process_rx_fcd(t30_state_t *s, const uint8_t *msg, int len)
                the frame, and let retries sort things out. */
             span_log(&s->logging, SPAN_LOG_FLOW, "Unexpected %s frame length - %d\n", t30_frametype(msg[0]), len);
         }
-        else
+        else
```
Comment (andywolk) [inline src/t30.c:3596]: Please cleanup what's left from the previous commit.

**Comment by andywolk** [comment]
@coppice-git Can you please formally approve the change Dragos did?

**Review by andywolk — APPROVED** [review]

**Pull request merged**